src/api-server.ts

```diff
@@ -271,17 +273,23 @@ async function createTopicWithRskAddress (call: any) {
     }
     console.log(`createTopicWithRskAddress ${JSON.stringify(call.request)} `)
     try {
-        console.log(`locatePeerID ${JSON.stringify(call.request.address)} `)
-        const address = await dht.getPeerIdByRskAddress(call.request.address);
-        console.log("address", address)
-        if (address === null) {
+        const rskAddress = call.request.address;
+        console.log(`locatePeerID ${JSON.stringify(rskAddress)} `)
+        const peerId = await dht.getPeerIdByRskAddress(rskAddress);
+        console.log("address", peerId)
+        if (peerId === null) {
             throw new Error("RSK Address Unknown");
         } else {
-            status = await subscribeToRoom(address);
-            streamConnectionTopic.set(address,call);
-            response = { address: address };
-            notificationMsg.channelPeerJoined.channel.channelId = address;
-            notificationMsg.channelPeerJoined.peerId = address;
+            status = await subscribeToRoom(peerId);
+            if (streamConnectionTopic.has(peerId)) {
+                streamConnectionTopic.get(peerId).set(rskAddress,call);
+            } else {
+                const rskAddresses = new Map([[rskAddress,call]])    
+                streamConnectionTopic.set(peerId,rskAddresses);    
+            }
+            response = { address: peerId };
+            notificationMsg.channelPeerJoined.channel.channelId = peerId;
+            notificationMsg.channelPeerJoined.peerId = peerId;
```

Comment:
I think we could use an object literal here. Also, Why is there both channelId and peerId, if they are the same?

Reply:
with CreateTopicWithRskAddress uses peerId as the channelID(topic)
They are the same for RSK

Reply:
Is there any situation when peerId is not the same as channelId?